pkgs/development/python-modules/fontmake/default.nix

```
@@ -1,40 +1,62 @@
 {
   lib,
   buildPythonPackage,
-  fetchPypi,
+  fetchFromGitHub,
+  gitUpdater,
+  pytestCheckHook,
+  fontmath,
+  fonttools,
   glyphslib,
+  setuptools,
   setuptools-scm,
+  skia-pathops,
+  ttfautohint-py,
   ufo2ft,
-  fonttools,
-  fontmath,
-  lxml,
-  setuptools,
+  ufolib2,
 }:
 
 buildPythonPackage rec {
   pname = "fontmake";
   version = "3.9.0";
   pyproject = true;
 
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-KrfT0fvE1fhaM2RH4LqRUda7yMHg2T59UdGi3SSZP7s=";
-    extension = "zip";
+  src = fetchFromGitHub {
+    owner = "googlefonts";
+    repo = "fontmake";
+    rev = "v${version}";
+    hash = "sha256-q6ul9MYbq85RpZE0ozHOCBNAR4r9InIjumadT1GyJ6k=";
   };
 
-  nativeBuildInputs = [ setuptools-scm ];
-
-  propagatedBuildInputs = [
-    glyphslib
-    ufo2ft
-    fonttools
-    fontmath
-    lxml
+  build-system = [
     setuptools
+    setuptools-scm
   ];
 
+  dependencies =
+    [
+      fontmath
+      fonttools
+      glyphslib
+      ufo2ft
+      ufolib2
+    ]
+    ++ fonttools.optional-dependencies.ufo
+    ++ fonttools.optional-dependencies.lxml
+    ++ fonttools.optional-dependencies.unicode;
+
+  optional-dependencies = {
+    pathops = [ skia-pathops ];
+    autohint = [ ttfautohint-py ];
+    json = ufolib2.optional-dependencies.json;
+    repacker = fonttools.optional-dependencies.repacker;
+  };
+
+  nativeCheckInputs = [ pytestCheckHook ] ++ optional-dependencies.autohint;
+
   pythonImportsCheck = [ "fontmake" ];
 
+  passthru.updateScript = gitUpdater { };
+
   meta = {
     description = "Compiles fonts from various sources (.glyphs, .ufo, designspace) into binaries formats (.otf, .ttf)";
     homepage = "https://github.com/googlefonts/fontmake";
```

Allows tools (e.g., `nix-update`) to add a link to the changelog to the commit message.

```suggestion
    homepage = "https://github.com/googlefonts/fontmake";
    changelog = "https://github.com/googlefonts/fontmake/releases/tag/v${version}";
```